Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #137548

update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

laithsakka committed:
add update_hint_regression and sum_floordiv_regression expected instruction count test

[ghstack-poisoned]

ezyang commented:
If we are not worried about causing hella merge conflicts, you could setup a bot to automatically commit updated thresholds. But this is also why I am a big fan of keeping this data not in the main repo...

ezyang commented on benchmarks/dynamo/pr_time_benchmarks/expected_results.csv:
```diff
@@ -7,3 +7,5 @@ basic_modules_ListOfLinears_eager,                         compile_time_instruct
 basic_modules_ListOfLinears_inductor_gpu_force_shape_pad,  compile_time_instruction_count, 16448960788,    0.015
 basic_modules_ListOfLinears_inductor_gpu,                  compile_time_instruction_count, 16403080126,    0.20
+update_hint_regression,                                    compile_time_instruction_count,  2433084132,    0.20
+sum_floordiv_regression,                                   compile_time_instruction_count, 1164734502,     0.015
```
I'm late to the party here, but please insert three lines of space between each of these, this will prevent Git from incorrectly merge conflicting if one person updates basic_modules_ListOfLinears_inductor_gpu and another updates update_hint_regression

laithsakka replied:
make sense

laithsakka commented:
sum floor div had two weired spikes idk if i will enable the regression test 
<img width="261" alt="Screenshot 2024-10-12 at 10 58 44 AM" src="https://github.com/user-attachments/assets/f6e0b113-ad30-40a0-9ba2-2d1122904398">
for it

laithsakka committed:
Update on "Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression"


update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression"


update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression"


update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
add 3 lines of space

laithsakka committed:
Update on "Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression"


update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression"


update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Enable failing diffs on update_hint_regression and sum_floordiv_regression and autograd benchmarks regression"


update_hint_regression has been behaving, so I am setting 2% noise threshold for it. 1.5% for sum_floordiv_regression. 

I have one concern, with the way we do the regression detection. small or changes <threshold level  will accumulate and eventually trigger failure. to avoid those would have to keep any eye on the dashboard and potentially refresh the expected result file regularly even when there is no faluires. .

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge -i
